src/g_lib/g_lib.cr

```diff
@@ -1,28 +1,52 @@
+require "time"
+
 require "../gobject"
 require_gobject "GLib"
 
 require "./*"
 require "../closure_data_manager"
 
 module GLib
-  def self.idle_add(&block : -> Bool)
+  enum Priority
+    Default      = LibGLib::PRIORITY_DEFAULT
+    Default_Idle = LibGLib::PRIORITY_DEFAULT_IDLE
+    High         = LibGLib::PRIORITY_HIGH
+    High_Idle    = LibGLib::PRIORITY_HIGH_IDLE
+    Low          = LibGLib::PRIORITY_LOW
+  end
+
+  def self.idle_add(priority : Priority = Priority::Default, &block : -> Bool)
     trampoline = LibGLib::SourceFunc.new { block.call ? 1 : 0 }
     LibGLib.idle_add(
-      LibGLib::PRIORITY_DEFAULT_IDLE,
+      priority,
       LibGLib::SourceFunc.new(trampoline.pointer, Pointer(Void).null),
       GObject::ClosureDataManager.register(trampoline.closure_data),
       ->GObject::ClosureDataManager.deregister
     )
   end
 
-  def self.timeout(seconds, &block : -> Bool)
+  def self.timeout(time : Time::Span, priority : Priority = Priority::Default, &block : -> Bool)
     trampoline = LibGLib::SourceFunc.new { block.call ? 1 : 0 }
-    LibGLib.timeout_add_seconds(
-      LibGLib::PRIORITY_DEFAULT_IDLE,
-      UInt32.new(seconds),
-      LibGLib::SourceFunc.new(trampoline.pointer, Pointer(Void).null),
-      GObject::ClosureDataManager.register(trampoline.closure_data),
-      ->GObject::ClosureDataManager.deregister
-    )
+    sourcefunc = LibGLib::SourceFunc.new(trampoline.pointer, Pointer(Void).null)
+    closure = GObject::ClosureDataManager.register(trampoline.closure_data)
+    if time.milliseconds
```

Comment:
This is just always truthy :)

Let's do

```cr
if time.to_i == time.total_seconds || time.total_milliseconds > UInt32::MAX
  timeout_add_seconds
elsif time.total_milliseconds > 0
  timeout_add
else
  raise ArgumentError.new("Timeout must be at least 1 millisecond")
end
```

And I agree we should also add `milliseconds_timeout` and `seconds_timeout` that always choose `timeout_add` and `timeout_add_seconds` respectively :)

Reply:
huh, i didn't realize that 0 was truthy. i think your idea looks a lot better, too

Reply:
Oh and as said we should throughoutly document this in a doc comment for this method :)